scripts/cmake/vcpkg_fixup_pkgconfig.cmake

```
@@ -274,60 +101,72 @@ function(vcpkg_fixup_pkgconfig)
-    debug_message("Release Files: ${_vfpkg_RELEASE_FILES}")
-    foreach(_file ${_vfpkg_RELEASE_FILES})
-        message(STATUS "Fixing pkgconfig file: ${_file}")
-        get_filename_component(PKG_LIB_SEARCH_PATH "${_file}" DIRECTORY)
-        file(RELATIVE_PATH RELATIVE_PC_PATH "${PKG_LIB_SEARCH_PATH}" "${CURRENT_PACKAGES_DIR}")
-        string(REGEX REPLACE "/$" "" RELATIVE_PC_PATH "${RELATIVE_PC_PATH}")
-        #Correct *.pc file
-        file(READ "${_file}" _contents)
-        string(REPLACE "${CURRENT_PACKAGES_DIR}" "\${prefix}" _contents "${_contents}")
-        string(REPLACE "${CURRENT_INSTALLED_DIR}" "\${prefix}" _contents "${_contents}")
-        string(REPLACE "${_VCPKG_PACKAGES_DIR}" "\${prefix}" _contents "${_contents}")
-        string(REPLACE "${_VCPKG_INSTALLED_DIR}" "\${prefix}" _contents "${_contents}")
-        string(REGEX REPLACE "^prefix[\t ]*=[^\n]*" "" _contents "${_contents}")
-        string(REGEX REPLACE "[\n]prefix[\t ]*=[^\n]*" "" _contents "${_contents}")
-        file(WRITE "${_file}" "prefix=\${pcfiledir}/${RELATIVE_PC_PATH}\n${_contents}")
-        unset(PKG_LIB_SEARCH_PATH)
-    endforeach()
-
-    if(NOT _vfpkg_SKIP_CHECK) # The check can only run after all files have been corrected!
-        foreach(_file ${_vfpkg_RELEASE_FILES})
-            vcpkg_fixup_pkgconfig_check_files("${PKGCONFIG}" "${_file}" "RELEASE" "${_vfpkg_SYSTEM_LIBRARIES}" "${_vfpkg_IGNORE_FLAGS}")
+    foreach(CONFIG RELEASE DEBUG)
+        debug_message("${CONFIG} Files: ${_vfpkg_${CONFIG}_FILES}")
+        if(VCPKG_BUILD_TYPE STREQUAL "debug" AND CONFIG STREQUAL "RELEASE")
+            continue()
+        endif()
+        if(VCPKG_BUILD_TYPE STREQUAL "release" AND CONFIG STREQUAL "DEBUG")
+            continue()
+        endif()
+        foreach(_file ${_vfpkg_${CONFIG}_FILES})
+            message(STATUS "Fixing pkgconfig file: ${_file}")
+            get_filename_component(PKG_LIB_SEARCH_PATH "${_file}" DIRECTORY)
+            if(CONFIG STREQUAL "DEBUG")
+                file(RELATIVE_PATH RELATIVE_PC_PATH "${PKG_LIB_SEARCH_PATH}" "${CURRENT_PACKAGES_DIR}/debug/")
+            else()
+                file(RELATIVE_PATH RELATIVE_PC_PATH "${PKG_LIB_SEARCH_PATH}" "${CURRENT_PACKAGES_DIR}")
+            endif()
+            # strip trailing slash
+            string(REGEX REPLACE "/$" "" RELATIVE_PC_PATH "${RELATIVE_PC_PATH}")
+            #Correct *.pc file
+            file(READ "${_file}" _contents)
+            string(REPLACE "${CURRENT_PACKAGES_DIR}" "\${prefix}" _contents "${_contents}")
+            string(REPLACE "${CURRENT_INSTALLED_DIR}" "\${prefix}" _contents "${_contents}")
+            string(REPLACE "${_VCPKG_PACKAGES_DIR}" "\${prefix}" _contents "${_contents}")
+            string(REPLACE "${_VCPKG_INSTALLED_DIR}" "\${prefix}" _contents "${_contents}")
+            string(REGEX REPLACE "^prefix[\t ]*=[^\n]*" "" _contents "${_contents}")
+            string(REGEX REPLACE "[\n]prefix[\t ]*=[^\n]*" "" _contents "${_contents}")
```

```suggestion
            string(REGEX REPLACE "(^|[\n])prefix[\t ]*=[^\n]*" "" _contents "${_contents}")
```

? I don't know if cmake's regex engine will be happy with that